internal/versions/versions.go

```diff
@@ -89,15 +92,38 @@ func (h *handler) V2ListComponentVersions(ctx context.Context, params operations
 		})
 }
 
+func (h *handler) hasArmAuthorization(ctx context.Context) bool {
+	armAuthorization := false
+
+	for _, releaseImage := range h.releaseImages {
+		if swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
+			err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
+			if err == nil {
+				armAuthorization = true
+			} else {
+				h.log.Debugf("Can't enable arm authorization, reason: %s", err.Error())
+			}
+			break
+		}
+	}
+	return armAuthorization
+}
+
 func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params operations.V2ListSupportedOpenshiftVersionsParams) middleware.Responder {
 	openshiftVersions := models.OpenshiftVersions{}
+	hasArmAuthorization := h.hasArmAuthorization(ctx)
```

Comment:
I think we can avoid this extra loop just by using a couple of booleans. let me know what you think:

```diff
diff --git a/internal/versions/versions.go b/internal/versions/versions.go
index c6c69983..410d3465 100644
--- a/internal/versions/versions.go
+++ b/internal/versions/versions.go
@@ -95,23 +95,20 @@ func (h *handler) V2ListComponentVersions(ctx context.Context, params operations
 func (h *handler) hasArmAuthorization(ctx context.Context) bool {
        armAuthorization := false
 
-       for _, releaseImage := range h.releaseImages {
-               if swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
-                       err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
-                       if err == nil {
-                               armAuthorization = true
-                       } else {
-                               h.log.Debugf("Can't enable arm authorization, reason: %s", err.Error())
-                       }
-                       break
-               }
+       err := auth.ValidateAccessToCPUArchitecture(ctx, h.authzHandler, common.ARM64CPUArchitecture)
+       if err == nil {
+               armAuthorization = true
+       } else {
+               h.log.Debugf("Can't enable arm authorization, reason: %s", err.Error())
        }
+
        return armAuthorization
 }
 
 func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params operations.V2ListSupportedOpenshiftVersionsParams) middleware.Responder {
        openshiftVersions := models.OpenshiftVersions{}
-       hasArmAuthorization := h.hasArmAuthorization(ctx)
+       checkedForArmAuthorization := false
+       hasArmAuthorization := false
 
        for _, releaseImage := range h.releaseImages {
                key := *releaseImage.OpenshiftVersion
@@ -120,8 +117,14 @@ func (h *handler) V2ListSupportedOpenshiftVersions(ctx context.Context, params o
                        *releaseImage.CPUArchitecture = common.DefaultCPUArchitecture
                }
 
-               if !hasArmAuthorization && swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
-                       continue
+               if swag.StringValue(releaseImage.CPUArchitecture) == common.ARM64CPUArchitecture {
+                       if !checkedForArmAuthorization {
+                               hasArmAuthorization = h.hasArmAuthorization(ctx)
+                               checkedForArmAuthorization = true
+                       }
+                       if !hasArmAuthorization {
+                               continue
+                       }
                }
 
                openshiftVersion, exists := openshiftVersions[key]

```